### ISSUE TITLE
[FR] Feature Request List

### ISSUE THREAD

**Issue by oldman4U** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1170]
Dear reader,

This Feature Requests list has three "functions":

1: It allows users to check, if a certain feature has already been requested, before starting a new ticket.

2: Mark features which are already under development to avoid, that several developers are working on the same function.

3: There is now a BOT which closes inactive tickets. Some of them are probably useful, so it makes sense to make them available at a more prominent place.

Notes:
Feel free to start a new FR Ticket, in case your feature or function is not listed here.
To join an existing FR, please click on the link below. Even the ticket is closed, it does not mean that it can not be used anymore.
Feel free to add your comments related to this FR ticket below.
New feature requests will be added as fast as possible.
Feature Requests shown in **Bold** are under development and if possible the developer is mentioned next to it.
Notes from the community are added to each FR in CAPITAL LETTERS.
Features which have been merged into the master branch, will be removed from this list.

General Features:
#290 Add a character generator in order to use klipper
#524 Vertical Screen GUI
**#634 The TFT display does not support the progress LED** PR #1209 available
#804 Transfer a file via Wifi ESP3D to the TFT-SD and/or TFT USB-Stick
**#1072 Smart RGB encoder control** PR #1209 available

Touchscreen Related Feature:
#296 Idle and back to main menu
#601 Invert filelist
**#732 Simplified Status Screen while Printing**
#763 Recover possibility to have the custom menu with icons
#810 Octoprint Print Status Screen
**#834 Filament change buttons** PR #1272 available 
**#865 Print finished dialog and jump back to Info Screen on TFT35**
**#1026 Alternate elapsed/remaining time during printing**
#1053 Option to deactivate the TFT LCD menu item
#1102 standard deviation on repeatability test
#1121 Runout Sensor On Board
#1137 Add ability to get to home screen with 1 click from any screen please
#1150 Include G34 Auto-Alignment to Movement/BedLevel UI
**#1166 Pre heat beep/tune/tone to acknowledge the set temperatures have been reached** PR #1209 available
**#1178 usb printing not able to control Neopixel PRINTER_EVENT_LEDS on tft35 v3 and poss other screens with usb drive ports** PR #1209 available
#1182 Show CHAMBER temperature alternately with Bed temperature
#1193 TFT 35 E3 v3.0 + skr 1.4 Power loss recovery Problem
#1211 Move X, Y, and Z Axes While Paused
#1238 Thumbnail visibility for Prusaslicer
#1264 Add GUI-support for G35 - Tramming assistant

Marlin Screen Related Features:
#301 Colour Mix menu
#594 Add button for Marlin Manual Mesh Bed Leveling 

Additional features and functions:
**#358 use #pragma once** PR #1209 available
#361 remote firmware deploy with build.py
#598 Implement "LittlevGL"
**#1115 Add option for temp sensors**
#1165 Smart Filament Sensor Module with Repetier-Firmware V2

Bug, Feature or Known Issue?:
#362 printing menu and echo
#438 OctoPrint and TFT comunication.
#571 Firmware assumes heated bed
#574 Optional' Enhancements (for CNC) Firmware assumes Tool, Extruder, Fan
#597 ESP32 CAM Compatibility
#958 WIFI REPETIER support

PS: A user asked me what happens with this list in case I do not want to maintain it anymore or when I die. Well, I am younger than my nickname makes you believe, but this is a valid question, so here is my answer.

Besides the possibility to reopen all tickets or to fall back to the status before this list has been made, there is still the option to copy and paste this ticket into a new one at any time. Or I am able to hand it over to another user, so I would do so...

Thank you

**Comment by traffic-light** [comment]
I think it's  a really good idea ;-)

You could keep it as a separate thing i think?

**Comment by oldman4U** [comment]
I hope BTT would add it to the pinned tickets.

Then we need a known issue.

Because yours is more a Frequently asked questions and how the report an
issue right documentation.

Peter van Weeterloo <notifications@github.com> schrieb am So. 11. Okt. 2020
um 19:39:

**Comment by oldman4U** [comment]
> I think it's a really good idea ;-)
> 
> You could keep it as a separate thing i think?
> 
> And maybe the possibility for everyone to tell what they are building? Don't think that is really needed because you set everything to bold. But it could be handy so that you don't need to change it every time when someone is doing a pr.

Hi Peter. Could you please delete your reply to make space for other replies, thank you.

**Comment by oldman4U** [comment]
> I think it's  a really good idea ;-)
> 
> 
> 
> You could keep it as a separate thing i think?
> 
> 
> 
> And maybe the possibility for everyone to tell what they are building? Don't think that is really needed because you set everything to bold. But it could be handy so that you don't need to change it every time when someone is doing a pr.

Hi Peter. Please delete your answer so I can release this ticket. 

Thank you

**Comment by oldman4U** [comment]
Hi @traffic-light. Could you please remove your comments we start with an empty ticket. Thank you

@bigtreetech Please pin this ticket, so I can ask all users with FR's to close their tickets.

Kind regards

**Comment by oldman4U** [comment]
@bigtreetech  Thank you BTT!!!

**Comment by JeremyKennedy** [comment]
What you are trying to do here goes against best practices for managing issues. Issues should remain open if they are unresolved, so that people who are searching for something can see if it has been resolved or not. You seem to think that having a single issue to contain other issues is a good idea maybe as an overview, but really that is what the main GitHub list of issues is for, this is completely unnecessary and confusing.

You should not be asking people to close unresolved issues, it is going to cause a ton of confusion. Issues should be standalone, rather than having some crazy "master issue" as you've created here. If you want to categorize feature requests so it is easier to find them, apply a label using GitHub's built-in functionality for that. Then you can search the issues list for that label, exactly as designed and intended by GitHub and all other issue trackers.

**Comment by oldman4U** [comment]
Hi Jeremy.

Thank you for taking the time to answer me. It seems that you know GitHub much better than me, so it would be great if you could convince BTT to add such a structure.

I am in the production and business industry since more than 25 years, half of the time as a product manager. For all our products there are documents like "Known Issues" and a "Knowledge Base" covering the most important topics at a single place. And most of the other products and developments I know have something similar.

It is also best practise, that someone starting a ticket is taking care of it and in case he is not interested anymore, he closes this ticket. In case it is unused for some time, usually the owner of the repo closes it, or a Bot. This is all not the case here. 

Many tickets are also opened, because the search function is not working well. There is a ticket #1158 which has "????" in its headline, search for it and check if you get a result. I get nothing.

Another reason is, that we had the situation that two engineers have been working on the same function "Mesh Bed Editor", without knowing from each other. My personal feeling is, that there are not too many engineers working on the software, so waisting time this way is far away from optimal.

Closing a ticket on page 2 or higher, and been listed in a pinned ticket instead is a good deal for me. The same time it makes the whole ticket system more useful. So I can see many benefits using this system for the price, that tickets which would never be found or which have not been used since months, have to be closed. Sounds like a good deal for me.

But lets see if we both agree on one thing. Look at this repository: https://github.com/bigtreetech/BIGTREETECH-SKR-V1.3/issues Do you think that unmanaged repositories are working good enough, so that actions like this are not needed. Once we agree that this is not the case and we find a better way to avoid this, I am willing to do what ever is needed to introduce this better system.


Again, I really appreciate your input.

Thank you

**Comment by traffic-light** [comment]
I agree with @JeremyKennedy .

But you have to take in mind that a lot of users here doesn't know how Github works. So it's not really handy to have a lot of tickets open with FR in it.

This way you keep the issues clean and hopefully more organized.

_The biggest problem in this project is the lack of project management._

**Comment by oldman4U** [comment]
I agree with him too. It is not the best solution but hopefully better than nothing. A few weeks ago we had around 250 open tickets and it was not usable anymore. Hope this helps to improve it further.

**Comment by traffic-light** [comment]
There is a PR (not finished yet) for 634 The TFT display does not support the progress LED

**Comment by oldman4U** [comment]
> There is a PR (not finished yet) for 634 The TFT display does not support the progress LED
> 
> 1209 ;-)

Updated. Sorry it took so long;-)

**Comment by traffic-light** [comment]
@oldman4U  it was not that long ago that I made it :-P

**Comment by Stellarspace** [comment]
> I agree with him too. It is not the best solution but hopefully better than nothing. A few weeks ago we had around 250 open tickets and it was not usable anymore. Hope this helps to improve it further.

Why won't @bigtreetech add a management bot or something?

**Comment by blueeagle69** [comment]
Feel free to add my https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/issues/1135 to the list 😀

**Comment by Stellarspace** [comment]
> Feel free to add my #1135 to the list 😀

It’s a bug, not a feature.

**Comment by blueeagle69** [comment]
> > Feel free to add my #1135 to the list 😀
> 
> It’s a bug, not a feature.

Thank you, I was aware it was a bug.
Or the other hand, it's something that isn't currently supported.

**Comment by SlackHacky** [comment]
Any update or clues to how to do on #524 Vertical Screen GUI ?

**Comment by oldman4U** [comment]
Hello SlackHacky.

Sorry for the late answer, has been a busy day...

As far as I know, there is currently no developer working actively on this feature. Are you an engineer which could develop the functionality, or are you a user who would like to use the screen in vertical direction?

**Comment by traffic-light** [comment]
@oldman4U my pr(#1209) resolves the next issues:

    resolves #634
    resolves #1072
    resolves #1178
    Resolves #358

**Comment by oldman4U** [comment]
> @oldman4U my pr(#1209) resolves the next issues:
> 
> ```
> resolves #634
> resolves #1072
> resolves #1178
> Resolves #358
> ```

GREAT! Thank you for letting me know. I will update the tickets asap.

**Comment by traffic-light** [comment]
@oldman thank you for your work!!!

**Comment by traffic-light** [comment]
I'll cover #1166 in my PR (#1209 )

**Comment by oldman4U** [comment]
Close this ticket now that it is pinned. Hopefully the Github app one day is able to show pinned tickets;-)

**Comment by readeral** [comment]
This is such a disaster @oldman4U

**Comment by oldman4U** [comment]
Hi. What is a disaster?

**Comment by readeral** [comment]
Consolidating issues like this. If you want something to list issues prioritise them, that's what Github Projects are for. Otherwise, duplicate projects get marked as duplicate and closed, while the original issue stays open. This is normal github etiquette.

**Comment by oldman4U** [comment]
How long are you watching THIS repository?

**Comment by readeral** [comment]
That's irrelevant, but if you truly want a dick-swinging contest in the realm of "how long" - you've been a user of github for only 12 months. Look into the projects feature, it's a far better approach than what you're trying to do.

**Comment by oldman4U** [comment]
Just to add.

Duplicate items are NOT deleted here, like the once which are orphaned for more than 1 year and where the owner does not answer anymore. PR's are waiting for weeks to get pushed and without the new bot we would have 350+ tickets. The owner of this repository does not answer tickets and .....

It is relevant, because we have no access and we can not make the things right without the support of the owner of this repo and because you have no glue how it "works" here - but only how it works in other repos.

Please feel free to improve it the way you think it is right. I do not say that this is the best way to do it, so please show me how it works.

Thank you

**Comment by readeral** [comment]
I've literally suggested that the Projects feature is how to go about it. Start here to learn: https://github.com/features#features-project-management

**Comment by oldman4U** [comment]
So you want to tell me what I have to do!?

Why you are not doing it? Are you too good for this?

**Comment by readeral** [comment]
I literally came to this repo from the Klipper repo to see how I might be able to contribute. I can't get a true sense of what's going on because a significant number of issues that are live issues are closed and compiled into this one and conversations are now dead. *shrug* If you really want a little kingdom, have at it.

**Comment by oldman4U** [comment]
I want to hand this kingdom over to someone who knows better, maybe someone like you!

What I do not want is someone new to THIS repo, knowing how it should be, seeing that it is not like this but NEVER tried to make it right.

Please stay here for more than one week and see by yourself and then let's discuss, ok?

**Comment by oldman4U** [comment]
Klipper support is highly requested and has nothing to do how this repo is used or administrated. So it would be great if you could take this over.

**Comment by traffic-light** [comment]
@readeral maybe you should know that not everyone knows how github works.
A lot of users here do not know anything. just placing their request and don't know how to search for it.
With this list it could be easier for users to find out if their FR is already there.

Yes. It's  far from perfect. But it's better then nothing.

I agree that FR's shouldn't be closed because there is a list. It should be closed trough a PR.

And I think the question: How long you're watching this repo is a good question. You're just showing in a few messages that you DON'T know anything how this repo is.

I'm not trying to scare you away from this repo. But you need to know more about the average user here.

**Comment by oldman4U** [comment]
Thank you.

3 Questions:
1. What makes you believe, that duplicated items are automatically deleted in this repository.
2. How can Project features be introduced into a repository as a new feature without being the owner of the same repository.
3. How is it going with the Klipper integration?

**Comment by oldman4U** [comment]
readeral seems to be a nine day wonder....

I delete my comments above to keep this ticket usable.

**Comment by traffic-light** [comment]
@oldman4U I think so... I'll remove my comments too

**Comment by readeral** [comment]
Or maybe it takes more than 9 days to wrap my head around a new code base in a language I have barely used. I'm more likely to make a PR after two months. This is the pace of open source.

**Comment by oldman4U** [comment]
Not talking about a PR but a short answer.

Alan Reader <notifications@github.com> schrieb am Mo. 14. Dez. 2020 um
21:24:

**Comment by readeral** [comment]
Clearly wasn't worth my effort to stroke your ego

**Comment by oldman4U** [comment]
Based on your introduction I expected not a lot, but even this was too
much.

Alan Reader <notifications@github.com> schrieb am Mo. 14. Dez. 2020 um
21:28:

**Comment by oldman4U** [comment]
@Stellarspace @blueeagle69 @SlackHacky

Could you please be so kind and close your comments above as a part of my housekeeping work.

Thank you and Merry Christmas

**Comment by radek8** [comment]
# 1298 On/Off terminalACK button
you can delete, it's done
Thank you

**Comment by oldman4U** [comment]
Hi radek.

Thank you for letting me know.

Happy printing

**Comment by bigtreetech** [comment]
@oldman4U Hello, Thank you very much for your help in the project. Do you have Facebook to communicate with? If it is not convenient to publish, you can send your contact information to our email “Tech@biqu3d.com”

**Comment by oldman4U** [comment]
@readeral 

I hope you are doing fine.

It is now more than 2 months ago where you not only enlighted us with your appearance, but where you also told us that this amount of time is also the usual timeframe for a PR in the OS community. Now I ask myself how much substance has been behind your comments here and I am sure, also the whole Klipper community can not wait to see what you have done for them.

So please let me know when you will show us your contribution to this project.

Thank you

**Comment by labotecno** [comment]
hello i get the G26 test leveling Working i need to finish some icon in other screen , how i can pull request ? [https://github.com/labotecno/BIGTREETECH-TouchScreenFirmware](url)

**Comment by oldman4U** [comment]
Sounds great!

I use GitHub Desktop for pushing PRs. Try it out and let us know in case you have a problem.

**Comment by labotecno** [comment]
i will add babystep menu when test leveling is enabled ! its working too ! like we can adjust offset z at the moment like prusa printer ! 

> 
> 
> Sounds great!
> 
> I use GitHub Desktop for pushing PRs. Try it out and let us know in case you have a problem.

ok i ill tell you soon thank you

**Comment by Domush** [comment]
This consolidated list does not replace feature requests. 

None of the original feature requests should be closed as none of them have been implemented.

Closing the linked feature requests betrays the entire purpose of GitHub and makes it **more** difficult to fulfill feature requests, as they can no longer be properly tracked.

All of the linked feature requests must be reopened in their original requests. No wonder development of these TFT's has slowed, nobody can issue pull requests because all of the linked feature requests are closed.

**Comment by oldman4U** [comment]
@Domush

why are you posting this here?

**Comment by traffic-light** [comment]
@oldman4U I think he's new at this repo. So he doesn't know why this is here.

**Comment by readeral** [comment]
He's posting it here because he actually understands GitHub

**Comment by oldman4U** [comment]
readeral

Are you here to present the results of your great work?

I am sure your friend Edward understand GitHub and you both are invited to improve this repo and make it a "real" GitHub.

So far all you and your friend left here is hot air. Not a single line of new code or a bug which could be fixed because of you.

**Comment by oldman4U** [comment]
@Domush 

I already tried to explain the situation here to your friend readeral before. Obviously he did not understand, so I'll try it again.

The time I joined this repository there have been almost 300 open tickets. Most of them have been not been updated for weeks and months, almost like in this repo: https://github.com/bigtreetech/BIGTREETECH-SKR-V1.3/issues

Unfortunately BIGTREETECH was too busy doing other things and therefore nobody was taking care. So some users like radek8, traffic-light,Tygrys-1, Gurmeet, and myself tried to organise this repo without the help of the owner. A few months later a bot was installed which closes tickets after 9 weeks of inactivity. To make sure that important tickets are still available without the need to search for closed tickets, two summary tickets have been opened.

To answer directly to your ticket above:

This consolidated list does not replace feature requests.
No, it does not. But what makes you believe that this could be the case?

None of the original feature requests should be closed as none of them have been implemented.
Agreed. It would be like this in an ideal world where the owner of this repo (or the user which opened a ticket) closes the ticket once it is not needed anymore. Unfortunately this is not the case.

Closing the linked feature requests betrays the entire purpose of GitHub and makes it more difficult to fulfill feature requests, as they can no longer be properly tracked.
Why does it make it more difficult to fulfill feature requests?
In an ideal world feature requests would be collected here, brought over to a development tool like JIRA once it is approved and closed automatically once fixed. But here it is sodom and gomorra, where bugs and features which are done are still in open tickets and vice versa and where users start tickets which are completely unrelated. See the SKR v1.3 or Marlin repo.

All of the linked feature requests must be reopened in their original requests. No wonder development of these TFT's has slowed, nobody can issue pull requests because all of the linked feature requests are closed.
Feel free to ask BTT, the owners of the tickets or someone else to open them again. I expect you to stay then here to take care - which would be great! We need people here actively working on this project. What we do not need here are users who tell others what to do and what is right and wrong, but not willing or able to contribute. Not sure I understand the need of an open feature request and the ability to issue a PR. Please explain.

I also have a question to you. It seems that you are a very smart developer with a successful history and I am sure you have a lot to do. What makes this little unimportant repository suddenly so important for you that you took the time to delight us with your presence?

Kind regards

oldman

**Comment by oldman4U** [comment]
Sorry, last question.

Do you really exist or are you a fake person because readeral has no friends?

7 repositories, with a total of 2 contributions in the last year and the domain on your profile: http://trulyfallacious.com can be bought.

An active GitHub user looks different. But I am sure you have several accounts, right!?

**Comment by readeral** [comment]
> readeral
> 
> Are you here to present the results of your great work?
> 
> I am sure your friend Edward understand GitHub and you both are invited to improve this repo and make it a "real" GitHub.
> 
> So far all you and your friend left here is hot air. Not a single line of new code or a bug which could be fixed because of you.

If you look at my github activity, you'll note it's almost entirely javascript. I'm not a full-time programmer, and so learning C and C++ sufficiently to contribute is going to take me some time. Hopefully I'll be able to contribute to fix my issue in 2021, but BOY would I LOVE someone who is proficient in C to have VISIBILITY OF ISSUES so they can just jump in and say "Oh yeah I can do that" but no, the way you've molested this repo has made that almost impossible.

**Comment by readeral** [comment]
Also, having looked at Domush's profile, see the "Pro" down the bottom there? That probably means they have been a github user for a while, and so paid to have private repositories (prior to them becoming free in the past year).

**Comment by oldman4U** [comment]
So this means that you can not contribute here because I messed up this repo so much. This is a massive lost for the community here but I understand your decision.

Still not sure how you and your friend  could even find time to come and visit us.

Good luck

**Comment by oldman4U** [comment]
In one of your useless posts you wrote you came over from Klipper. You have 42 repos but Klipper is not there and I checked the tickets and PR's at the Klipper GitHub and it seems that you have been very active there because the result is 0.

You are my hero and where has you friend Domush been the last couple of hours? Another 9 day wonder?

**Comment by Domush** [comment]
Where else should it be posted? It's literally a comment on this topic, so it belongs here.


On Mar 2, 2021 7:52 PM, oldman4U <notifications@github.com> wrote:
@Domush

why are you posting this here?

**Comment by Domush** [comment]
Admins here should really consider whether or not they want a toxic person like @oldman4U chasing away developers. Looking at the pull requests it looks like you're down to only a couple of contributors remaining.On Mar 2, 2021 7:55 PM, oldman4U <notifications@github.com> wrote:
So this means that you can not contribute here because I messed up this repo so much. This is a massive lost for the community here but I understand your decision.
Still not sure how you and your friend  could even find time to come and visit us.
Good luck

**Comment by oldman4U** [comment]
Feel free to talk to BTT and take over. So far you did for this repo as much as your useless friend.

**Comment by readeral** [comment]
I honestly don't know why you call me 'useless friend' given I've never met Domush. BTT really have shot themselves in the foot allowing you access. But they clearly don't care enough anyway.

**Comment by readeral** [comment]
@Domush I'd love to see your fork

**Comment by filip-smid** [comment]
> @Domush I'd love to see your fork

Everyone would at this point. 70k lines of could would change a lot, but alas, no proof. so these are just puff dragon screams for now.

**Comment by Domush** [comment]
> @Domush I'd love to see your fork

I'll invite you to the repository later today.

**Comment by oldman4U** [comment]
Respect Domush. At least you delivered something and I am sure it can be useful for certain users. Unfortunately I can not check it because 3D prints are all I do on my machines.

But you readeral, you are really useless. Where is the Klipper support you have been talking about more than 2 months ago? All you added to this community so far is hot air and that you wasted my time. If this is everything we can expect from you I would recommend that you to try it somewhere else, because I have more important things to do than to educate you.

**Comment by bttguy** [comment]
@oldman4U Can you add#1400 RRF compatibility for the FR list? :) Ther were already some commits to start RRF support, maybe this FR list could help to attract more developers. Thank you in advance!

**Comment by oldman4U** [comment]
Hi,

Yes sure.

Hope this helps;-)

**Comment by bttguy** [comment]
Thank you @oldman4U ! I am also a RRF user, but can't code, bit try my best to ask for help in the RRF community. Do you think it is a good idea to add the RRF relates changes and requirements to the Readme along with the already implemented features? I will check out how I can commit readmes, but I could take my part in the job with documentation, to make it more visible.

**Comment by bttguy** [comment]
Thanks @oldman4U I have a Windows, but I search for some compatible editor and experiment with a fork how GitHub works apart from chatting with people :D

**Comment by sarvenn** [comment]
@bigtreetech hi,
Could you please review these two topics?
Can they be added to the TO DO list?
#1506 Clickable status buttons during print
#1572 Optimised Print menu and List/thumbnail selection addition to top row

**Comment by oldman4U** [comment]
sarvenn.

They are already on the list, check again.

**Comment by sarvenn** [comment]
> sarvenn.
> 
> They are already on the list, check again.

Hi @oldman4U ,
I meant here #1830 .

**Comment by CyberDeck** [comment]
@oldman4U, I will try to come up with some ideas for #1896. Not so much for a child lock (a 3D printer should be always put in a safe place where no child could reach it) but more in terms of a general "accidential touch" / "rotate" lock like a mobile keylock feature.

First idea I will give a try: Implement some kind of "long touch" on the title bar to lock and unlock the screen. Should not be to hard to implement.

**Comment by digant73** [comment]
@oldman4U FR #1945 was implemented so you can remove it from the list

**Comment by oldman4U** [comment]
👍🏻

Antonino Di Guardo ***@***.***> schrieb am Sa. 31. Juli 2021
um 14:33:

**Comment by digant73** [comment]
@oldman FR #438 and FR #810 have been implemented so they can be removed from the list.
Furthermore the readme file should be updated reporting the following Octoprint integration guide

**INTEGRATION WITH OCTOPRINT / OCTOPRINT TRIGGERING COMMANDS:**
Octoprint can optionally trigger some actions to the TFT sending specific gcodes. The following actions and the related triggering gcodes are currently supported by the TFT fw:

**start:**
M118 A1 P0 action:print_start
**end:**
M118 A1 P0 action:print_end
**cancel:**
M118 A1 P0 action:cancel
**pause:**
M118 A1 P0 action:pause
**resume:**
M118 A1 P0 action:resume
**time remaining progress:**
M118 A1 P0 action:notification Time Left <XX>h<YY>m<ZZ>s (e.g. 02h04m06s)
M117 Time Left hms
**file data progress:**
M118 A1 P0 action:notification Data Left <XXXX>/<YYYY> (e.g. 123/12345)
M117 Data Left <XXXX>/<YYYY>

When the trigger "print_start" is received, the TFT switches to Printing menu.
Once on Printing menu, the "pause", "resume" and "stop" buttons on the menu will be disabled.
That means only Octoprint will control the print.
Only on print end or cancel the TFT Printing menu is finalized (statistics available etc...) and unlocked (it can be closed).

**NOTE:** A new plugin on Octoprint implementing the above protocol should be the preferable way (available to everyone)

**Comment by kisslorand** [comment]
> #2005 MULTI_VOLUME Support

Done in PR #2418, it all depends on Marlin's repository maintainers. (https://github.com/MarlinFirmware/Marlin/pull/23780)

**Comment by Tannoo** [comment]
Could we get all marlin mode options in the TFT?

Would be great to get rid of marlin mode and it's cables.

**Comment by radek8** [comment]
> Could we get all marlin mode options in the TFT?
> 
> Would be great to get rid of marlin mode and it's cables.

And what specifically are you missing? :-)

**Comment by Tannoo** [comment]
Status updates during PID tuning.

Probe offset wizard

Those are off the top of my head that I needed Marlin Mode for so far.

I will let you know when I find more.

**Comment by radek8** [comment]
As far as I know, both named functions are available in touch mode (in the current version).

**Comment by radek8** [comment]
Special functions for atypical printer configurations can still be performed via commands in a terminal window.
I haven't run the Marlin mode on the printer for over a year. 
Everything I need is configurable via touch mode.
I think in most cases with higher control comfort.

**Comment by Tannoo** [comment]
Status updates during PID tuning.

Probe offset wizard

Those are off the top of my head that I needed Marlin Mode for so far.

I will let you know when I find more.

**Comment by Tannoo** [comment]
Unless these are tucked away that I have not found.

**Comment by Tannoo** [comment]
Is there a V4 in the pipes?

**Comment by radek8** [comment]
![20240105_211548](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/assets/46979052/0836d02a-ef06-4b96-8fc3-a88027ffb7e1)

**Comment by Tannoo** [comment]
Okay. You got me ;)
